.github/workflows/ci.yml

```diff
@@ -1,311 +1,177 @@
+    name: Leaked Credential Test
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        platform: [ windows-x64, macos-x64, macos-arm64, linux-x64, linux-musl-x64, linux-armv7, linux-arm64 ]
-
     steps:
-      - uses: actions/checkout@v2
-      - name: Set Vars
-        run: |
-          echo "GITHUB_SHA_SHORT=${GITHUB_SHA::7}" >> $GITHUB_ENV
-          echo "GITHUB_REF_SHORT=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
-
-      - name: Build distributable
-        uses: docker/build-push-action@v2
-        with:
-          file: build-scripts/Dockerfile.${{ matrix.platform }}
-          outputs: dist/${{ matrix.platform }}
-          build-args: |
-            STACKS_NODE_VERSION=${{ github.event.inputs.tag || env.GITHUB_SHA_SHORT }}
-            GIT_BRANCH=${{ env.GITHUB_REF_SHORT }}
-            GIT_COMMIT=${{ env.GITHUB_SHA_SHORT }}
-          
-      - name: Compress artifact
-        run: zip --junk-paths ${{ matrix.platform }} ./dist/${{ matrix.platform }}/*
-
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: ${{ matrix.platform }}
-          path: ${{ matrix.platform }}.zip
-
-  call-docker-platforms-workflow:
-    if: ${{ github.event.inputs.tag != '' }}
-    uses: stacks-network/stacks-blockchain/.github/workflows/docker-platforms.yml@master
+      - name: Checkout the latest code
+        id: git_checkout
+        uses: actions/checkout@v3
+      - name: TruffleHog Scan
+        id: trufflehog_check
+        uses: trufflesecurity/trufflehog@main
+        with:
+          path: ./
+          base: ${{ github.event.repository.default_branch }}
+          head: HEAD
+
+  ###############################################
+  ## Build Tagged Release
+  ###############################################
+  # Release Tests
+  #   Only run if (a tag was passed in) -AND- (we're building the main branch which isn't a PR)
+  # Release Tests
+  # Only run if:
+  #    - tag is provided
+  #    - AND
+  #        - this is the default branch
+  #        - AND Not a PR
+  release-tests:
+    if: ${{ inputs.tag != '' && (github.ref == format('refs/heads/{0}', github.event.repository.default_branch) && !contains(github.ref, 'refs/pull')) }}
+    name: Release Tests
+    uses: stacks-network/stacks-blockchain/.github/workflows/stacks-blockchain-tests.yml@master
+
+  # Build source binaries
+  # Only run if:
+  #    - tag is provided
+  #    - AND
+  #        - this is the default branch
+  #        - AND Not a PR
+  build-source:
+    if: ${{ inputs.tag != '' && (github.ref == format('refs/heads/{0}', github.event.repository.default_branch) && !contains(github.ref, 'refs/pull')) }}
+    name: Build Binaries
+    uses: stacks-network/stacks-blockchain/.github/workflows/build-source-binary.yml@master
```

Comment:
A fair point - I did consider that route at first, but ultimately i decided i didn't want to change the process of how we build things too drastically with this PR. most of the workflows are nearly identical to the existing workflow, just cleaned up a bit to make it easier to address future changes. 

The second reason i chose not to go that route was because then there would be another repo that would have to be maintained and I didn't want to add more work (objectively) to the changes here. 

The main focus I had was to first build the arm64 docker image, and the second was to perform a bit of a cleanup. 
I added in clippy and the dependency actions as possible nice to haves since they've come up in conversations. 

The next iteration of CI for this repo could absolutely use composite actions as you've suggested, I simply felt it was a bridge too far considering the goals I set out with. "don't rock the boat *too* hard"

Reply:
@wileyj Just an FYI if considering composite actions in the future, the actions can actually be stored in this repo like the re-usable workflows are. They'd just be in a different directory. [See here for an example in the stacks-web-wallet repo.](https://github.com/hirosystems/stacks-wallet-web/blob/dev/.github/actions/provision/action.yml)

Reply:
Yup - I think long term though, it would make more sense to move them out of this repo since getting a workflow change added here is harder than i think it should be. 
with some external composite actions, it would be easier to make quicker iterations and get them active. we can discuss later though, just some initial thoughts.